language/vm/src/file_format.rs

```diff
@@ -373,14 +373,15 @@ pub struct FieldDefinition {
 
 /// A `FunctionDefinition` is the implementation of a function. It defines
 /// the *prototype* of the function and the function body.
+
 #[derive(Clone, Debug, Default, Eq, PartialEq)]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
 #[cfg_attr(any(test, feature = "fuzzing"), proptest(params = "usize"))]
 pub struct FunctionDefinition {
     /// The prototype of the function (module, name, signature).
     pub function: FunctionHandleIndex,
-    /// Flags for this function (private, public, native, etc.)
-    pub flags: u8,
+    /// Flag to indicate if this function is public.
+    pub is_public: bool,
```

Comment:
I would actually leave the flag as it was instead of changing that to a bool. 
We can do it later if you want but it just seems a waste to have a bool for effectively a flag that can evolve to more, who knows...

Reply:
Hmm... I see where you are coming from. From my view the old flag is a compact representation of multiple boolean flags and that compression can happen at serialization time. An analogy would be that we use u16 instead of uleb in the rust repr for indices. Having a bool won't stop you from adding a new flag without breaking backward compatibility -- you'll need to change the rust representation but the actual Move binary format will still have the bits reserved which you can use for the new flags. 

To me the struggle is that the CodeUnit is now optional -- so if we want the bit that controls whether the function is native be part of the flag, we'll need to merge the two at serialization time, which is a little bit weird to me so I just decided to take the whole thing apart. I also found it annoying that any client using the file format has to understand what the bit masks mean. Although as we discussed, maybe it's because those clients shouldn't be using the file format directly and that we really should make the file format as close to bare metal as possible. Anyways I personally found bools more suitable for our current clients, but let's keep the conversation open and revisit later.